platform/marvell-arm64/platform.conf

```diff
@@ -174,6 +294,77 @@ get_install_device()
 
 create_demo_partition() {
 
+    if [ "$PLATFORM" = "$PLATFORM_AC5X_SUPERMICRO_G3748" ]
+    then
+        onie_part=$(sgdisk -p "$blk_dev" | grep -e "$onie_part_label" | awk '{print $1}')
+        sonic_part=$(sgdisk -p "$blk_dev" | grep -e "$sonic_part_label" | awk '{print $1}')
+        swap_part=$(sgdisk -p "$blk_dev" | grep -e "$swap_part_label" | awk '{print $1}')
+        debug_part=$(sgdisk -p "$blk_dev" | grep -e "$debug_part_label" | awk '{print $1}')
+
+        if [ -z "$onie_part" ]
+        then
+            echo "Partitioning is Incorrect. Re-partitioning"
+            sgdisk -oZ ${blk_dev}
+
+            sgdisk --new ${onie_part_num}:0:+${onie_part_size} \
+                --change-name=${onie_part_num}:${onie_part_label} \
+                --typecode=${onie_part_num}:8300 -p ${blk_dev}
+
+            if [ $? -ne 0 ]; then
+                echo "Failed creating onie eMMC partition"
+                return 2
+            fi
+
+            mkfs.ext4 -F -L ${onie_part_label} ${blk_dev}p${onie_part_num}
+        else
+            fsck.ext4 -v -p ${blk_dev}p${onie_part_num}
+        fi
+
+        if [ -z "$debug_part" ]; then
+            # Create debug partition
+            sgdisk --new ${debug_part_num}:-${debug_part_size}: \
+                --change-name=${debug_part_num}:${debug_part_label} \
+                --typecode=${debug_part_num}:8300 -p ${blk_dev}
+
+            mkfs.ext4 -F -L ${debug_part_label} ${blk_dev}p${debug_part_num}
+        else
+            fsck.ext4 -v -p ${blk_dev}p${debug_part_num}
+        fi
+
+        if [ -z "$swap_part" ]; then
+            # Create swap partition
+            swap_start_num=-${debug_swap_part_size}
+            swap_end_num=-${debug_part_size}
+            #sgdisk --new ${swap_part_num}:-${swap_part_size}: \
+            sgdisk --new ${swap_part_num}:${swap_start_num}:${swap_end_num} \
+                --change-name=${swap_part_num}:${swap_part_label} \
+                --typecode=${swap_part_num}:8200 -p ${blk_dev}
+        fi
+
+        if [ -z "$sonic_part" ]; then
+            # Create sonic partition
+            start_num=$(sgdisk -F $blk_dev)
+            end_num=$(sgdisk -E $blk_dev)
+
+            sgdisk --new ${sonic_part_num}:${start_num}:${end_num} \
+                --change-name=${sonic_part_num}:${sonic_part_label} \
+                --typecode=${sonic_part_num}:8300 -p ${blk_dev}
+
+            mkfs.ext4 -F -L ${sonic_part_label} ${blk_dev}p${sonic_part_num}
+        else
+            mkfs.ext4 -F -L ${sonic_part_label} ${blk_dev}p${sonic_part_num}
+        fi
+
+        partprobe
+        sync
+
+        mkswap ${blk_dev}p${swap_part_num}
+        echo "${blk_dev}p${swap_part_num}    swap  swap  defaults   0   0" > /etc/fstab
+        swapon -a
+
+        return 0
+    fi
```

Comment:
this seems complicated than existing. first, we do not use swap in sonic, if we do want to support swap in sonic, this should be a generic feature, not platform specific.